src/draw.rs

```diff
@@ -0,0 +1,81 @@
+use crate::color::ARGB;
+use crate::pixel::{PixelArray, PixelArrayMut};
+
+#[inline]
+fn is_inside_circle(x: isize, y: isize, r: isize) -> bool {
+    (x - r).pow(2) + (y - r).pow(2) < r.pow(2)
+}
+
+#[inline]
+fn border_color(color: ARGB) -> u32 {
+    if color.is_dark() {
+        ARGB::WHITE.into()
+    } else {
+        ARGB::BLACK.into()
+    }
+}
+
+pub fn draw_magnifying_glass(
```

Comment:
This is where the picker is drawn -> it's a one-dimensional array of ARGB values.
I tried to keep it as simple as possible, and hopefully it's clear to understand what it's doing.